docs/source/raft_configuration.md

```diff
@@ -330,5 +330,29 @@ channels (this does not mean they share data --- they do not!), if a Raft node i
 part of the consenter set in many channels, you might want to lengthen the amount
 of time it takes to trigger an election to avoid inadvertent leader elections.
 
+2. As aforementioned, write-ahead logs (wal) and snapshots should be persisted in
+order to attain safety property of Raft protocol. However, it may happen if a node
+loses its data by accident, and the node would panic with following message:
+```
+tocommit(X) is out of range [lastIndex(Y)]. Was the raft log corrupted, truncated, or lost?
+```
+Manual operation would be required to resurrect problematic node in this case.
```

Comment:
I would break this section into two perhaps?

Safe recovery:
The only way to handle the corruption or loss of a WAL is to remove the corrupted consenter from the quorum, and add a new one in its place.  ...

Unsafe recovery: 

It is possible to recover from a WAL corruption without removing the consenter from the consenter set and re-adding it.  However, these procedures violate the Raft protocol safety guarantees and, in certain unlikely cases, could result in a blockchain fork.  To minimize these possibilities, ensure that your cluster has quorum on all channels for a few minutes before proceeding. (I think this is good enough? We just need a quorum to advance the index via heartbeats past any possible conflicting index in the broken node's WAL, right?)

Option 1: ...

Option 2: ...